sherpa/models/tests/test_template_unit.py

```diff
@@ -47,13 +77,478 @@ def test_309(make_data_path):
+                          (1, 39, 40),
+
+                          (2, 11, weighted([(0.9, 5), (0.1, 15)])),
+                          (2, 15, weighted([(0.5, 5), (0.5, 15)])),
+                          (2, 27, weighted([(0.3, 15), (0.7, 20)])),
+                          (2, 31, weighted([(0.9, 20), (0.1, 40)])),
+                          (2, 39, weighted([(0.1, 20), (0.9, 40)])),
+
+                          # Weights are 1/<distance>
+                          (3, 11, weighted([(1/1, 5), (1/9, 15), (1/19, 20)])),
+                          (3, 15, weighted([(1/5, 5), (1/5, 15), (1/15, 20)])),
+                          (3, 27, weighted([(1/7, 15), (1/3, 20), (1/13, 40)])),
+                          (3, 31, weighted([(1/11, 15), (1/1, 20), (1/9, 40)])),
+                          (3, 39, weighted([(1/9, 20), (1/1, 40), (1/11, 44)])),
+                          ])
+def test_interpolate_par1(order, k, pval, expected):
+    """Check KNNInterpolator/Template distance calculation: 1 par
+
+    The idea is that we use a "constant" model so we can know what
+    the interpolated value should be.
+
+    I was surprised that order=1, 2, 3 all gave the same results,
+    but we only have one parameter so there's no real "power"
+    for the order value to address.
+
+    """
+
+    x = np.asarray([2, 4, 6])
+    templates = []
+    for scale, name in zip([5, 15, 20, 40, 44], ["m1", "m2", "m3", "m4", "m5"]):
+        template = TableModel(name)
+        template.load(x, np.ones_like(x) * scale)
+        templates.append(template)
+
+    tmpl = create_template_model("bob", ["pa"], np.asarray([[10], [20], [30], [40], [50]]),
+                                 templates)
+    assert isinstance(tmpl, Template)
+    assert tmpl.k == 2
+    assert tmpl.order == 2
+
+    # The x axis values don't really matter here, but keep them inside
+    # the original x=2, 4, 6 setting. A single value could be used,
+    # but include 2 just in case.
+    #
+    tmpl.order = order
+    tmpl.k = k
+    tmpl.pa = pval
+    assert tmpl([3, 5]) == pytest.approx([expected, expected])
+
+
+def check3(p1, p2, p3):
+    """What do we expect the answer to be?"""
+    x = np.asarray([2, 4, 6])
+    return p1 * x * x + p2 * x + p3
+
+
+@pytest.mark.parametrize("order,k,pval1,pval2,pval3,expected",
+                         # First checks are linear interpolation along
+                         # p3 axis at points where we have p1 and p2.
+                         #
+                         [(1, 2, 2, 3, 25, check3(2, 3, 25)),
+                          (2, 2, 2, 3, 25, check3(2, 3, 25)),
+                          (3, 2, 2, 3, 25, check3(2, 3, 25)),
+                          (1, 2, 0.5, 1, 15, check3(0.5, 1, 15)),
+                          (2, 2, 0.5, 1, 15, check3(0.5, 1, 15)),
+                          (3, 2, 0.5, 1, 15, check3(0.5, 1, 15)),
+
+                          # Actual vaues are [39, 57, 81]
+                          # Why does the order setting not matter?
```

Comment:
In the implementation, k is not the "order", but the number of closet points that is taken into account. I still don't fully understand, why the setting does not matter here (it indicates that there is something still wrong), but I think this setting will have more of an effect with more "dimentions".